### ISSUE TITLE
ESP8266

### ISSUE THREAD

**Issue by iLLiac4** [sticilface/Adalight-ESP8266#1]
Can you please add some pictures of your setup?
Maybe i do not get it right but are you using serial or wifi to get the 'stream' to the ESP8266?

**Comment by sticilface** [comment]
quoted from the first line of the readme....
`This sketch will take GRB pixels over serial sent by hyperion` it does nothing else.  

This sketch does not function in any way with wifi, unless you add it.  i've not got a picture to hand, but you will need a USB to SERIAL connecter for it to work.  You will have to connect it correctly to the RX and TX of the ESP, and the USB connection to the computer running hyperion.  I have only tested this with HYPERION running on rasplex.  hyperion does not support UDP or wifi streaming, that i know of.  

see here for what is supported
https://github.com/tvdzwan/hyperion/wiki

I hope this helps.

If you have an alternative method of streaming ADALIGHT protocol over serial then you can use that.  It forms the same basic structure of sending bytes A D A hi lo chk followed by 3 bytes representing the three colours of a single RGB LED.  In this case you have to send them G R B.  again you can change this in the code if you wish.  see here for more info on adalight

https://github.com/adafruit/Adalight/blob/master/Arduino/LEDstream/LEDstream.pde

**Comment by iLLiac4** [comment]
Aha ok. I thought that it works wireless. Than it is basically just an alternative to Arduino Nano or. Teensy.

**Comment by sticilface** [comment]
yes, the exception being that you can put this function in the loop for example, and the ESP will handle other things at the same time.  Or put it in a switch and have it do other stuff not adalight.  it is non-blocking, no while loops waiting for serial etc.  This is important on the ESP for the WIFI for example, although no demonstrated here.  Other examples of the Adalight do not do this....

I also have a UDP function, that takes RGB pixels, but hyperion does not support that.

**Comment by penfold42** [comment]
This works great for me with OSMC and a RPi2.
The native GPIO DMA PWM code still seems flaky on my RPi2 AND I lose analog audio out at the same time. I also found the arduino board even with 500000 BPS to be quite slow.

I considered writing my own using the SPI interface, but this works great.

Have you considered SPI support as well? I wonder if that has less load on the RPi than USB or even the onboard UART.

**Comment by sticilface** [comment]
I've not considered that actually until you mentioned it.  I've not worked with SPI before.  could be easier..  my implementation struggles with the new ESP8266 hardware serial for some reason... you have to disable it when no receiving data.

**Comment by penfold42** [comment]
It's probably not worth adding SPI now. 

I wrote a driver for ws2811 style leds using SPI directly on the Pi

**Comment by sticilface** [comment]
thats cool.  got a link to it.  i like to keep the esp doing the heavy lifting as I have the LEDs controlled using MQTT,alexa, HTTP as well. :)
